lab-dan/model/storage.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+const objectConstructor = require('./objectConstructor')
+
+let storage = {}
+storage.data = {
+  '1234-test-obj': {
+    id: '1234-test-obj',
+    creationDate: new Date(12/11/2016),
+    name: 'Test',
+    breed: 'Shiba',
+  },
+}
+
+// let fs = blueBird.promisifyAll(require('fs'), { suffix: 'Promised'})
+
+// This is what a monkeyfied fs function looks like
+// fs.readFilePromised('file').then()
+// fs.writeFilePromised('file').then()
+
+storage.fetchItem = function(id) {
+  // return fs.readFilePromised(`${__dirname}/../data/data.json`)
+  //   .then(data => {
+  //     data = JSON.parse(data.toString())
+  //     return data[id]
+  //   })
+  if (!id) {
+    return storage.fetchAll()
+  } else if (typeof storage.data[id] === 'object') {
+    return Promise.resolve({code: 200, data: storage.data[id]})
+  }
+  return Promise.reject('not found\n')
+}
+
+storage.fetchAll = function() {
+  return Promise.resolve({code: 200, data: storage.data})
+}
+
+storage.postItem = function(obj) {
+  let newItem = new objectConstructor(obj.name, obj.breed)
+  storage.data[newItem.id] = newItem
+  return Promise.resolve({code: 201, data: storage.data[newItem.id]})
+}
+
+storage.putItem = function(obj) {
+  // as per the rules of put -- update existing object or create a new object
+  if (typeof storage.data[obj.id] === 'object') {
+    Object.keys(obj).forEach(key => {
+      storage.data[obj.id][key] = obj[key]
+    })
+    return Promise.resolve({code: 200, data: storage.data[obj.id]})
+  } else {
+    return storage.postItem(obj)
+  }
+}
+
+storage.deleteItem = function(id) {
```

Comment:
Given that you're working in a try/catch, you don't necessarily need to validate that the ID exists. Just let the try/catch handle it for you.